### PR TITLE
Bump manif version to 0.0.4

### PR DESCRIPTION
**Pull request by GiulioRomualdi**

### CODE REVIEW

**Review by traversaro — APPROVED** [review]

**Comment by GiulioRomualdi** [comment]
Even if the new version was released the CMake version didn't changed 
https://github.com/artivis/manif/blob/0d42c2de1a34628ca9044b1f2e4dd17bbfead904/CMakeLists.txt#L4

**Comment by GiulioRomualdi** [comment]
Associated issue: https://github.com/artivis/manif/issues/232

**Comment by traversaro** [comment]
Can we drop the version check 0.0.4 for now so we can actually work with the 0.0.4 version that advertise itself still has 0.0.3 ?

File: .github/workflows/ci.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@ env:
   Catch2_TAG: v2.13.4
   Qhull_TAG: 2020.2
   CasADi_TAG: 3.5.5.2
-  manif_TAG: c8b4f16ea4d64e8d145aaf50bdbe31c756edb5b2
+  manif_TAG: 0.0.3
```
Comment (traversaro) [inline .github/workflows/ci.yml:20]: Still 0.0.3 ?

Comment (GiulioRomualdi) [reply .github/workflows/ci.yml]: https://github.com/dic-iit/bipedal-locomotion-framework/pull/339/commits/9bded1461834b5cec7b746f662dd738eedcbf5d1

**Pull request merged**